### PR TITLE
calendar---created/updated events

### PR DESCRIPTION
**Pull request by invaliduser**
adds calendar_event_created and calendar_event_updated events, and tests for each

### CODE REVIEW

File: src/transformer/events/core/badge_revoked.php
Comment (milt) [inline src/transformer/events/core/badge_revoked.php:1]: What's this doing here?

File: src/transformer/events/core/calendar_event_created.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Transform for course viewed event.
+ *
+ * @package   logstore_xapi
+ * @copyright Jerret Fowler <jerrett.fowler@gmail.com>
+ *            Ryan Smith <https://www.linkedin.com/in/ryan-smith-uk/>
+ *            David Pesce <david.pesce@exputo.com>
```
Comment (milt) [inline src/transformer/events/core/calendar_event_created.php]: Did any of these people write this?

File: src/transformer/events/core/calendar_event_created.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        ];
+    }
+
+    $encoded =json_encode($statements);
+    /*    echo <<<END
+             <script type="text/javascript">
+             var s = $encoded;
+             console.log(s);
+             </script>
+             END;*/
```
Comment (milt) [inline src/transformer/events/core/calendar_event_created.php]: please work on getting dev/debug artifacts out of here before PR time

File: src/transformer/get_event_function_map.php [hunk 3]
```diff
@@ -35,6 +35,9 @@ function get_event_function_map() {
     $availableevents = [
         '\core\event\badge_awarded' => 'core\badge_awarded', //'debug\debug_event',
         '\core\event\badge_revoked' => 'core\badge_revoked', //'debug\debug_event'
+        '\core\event\calendar_event_created' => 'core\calendar_event_created',
+        '\core\event\calendar_event_updated' => 'core\calendar_event_updated',//'debug\debug_event',//,
+        '\core\event\calendar_event_deleted' => 'core\calendar_event_deleted',
         '\core\event\course_completed' => 'core\course_completed',
         '\core\event\course_viewed' => 'core\course_viewed',
         '\core\event\user_created' => 'core\user_created',
```
Comment (milt) [inline src/transformer/get_event_function_map.php]: remove commented out stuff

File: src/transformer/get_event_function_map.php [hunk 4]
```diff
@@ -96,10 +99,8 @@ function get_event_function_map() {
         '\mod_url\event\course_module_viewed' => 'mod_url\course_module_viewed',
         '\mod_wiki\event\course_module_viewed' => 'all\course_module_viewed',
         '\mod_workshop\event\course_module_viewed' => 'all\course_module_viewed',
-        '\totara_program\event\program_assigned' => 'totara_program\program_assigned',
-        '\core\event\calendar_event_created' => 'core\calendar_event_created',
-        '\core\event\calendar_event_updated' => 'core\calendar_event_updated',
-        '\core\event\calendar_event_deleted' => 'core\calendar_event_deleted',
```
Comment (milt) [inline src/transformer/get_event_function_map.php:103]: uh oh, were these left in on the previous merge?

File: tests/core/calendar_event_created/user_created_calendar_event/data.json
Comment (milt) [inline tests/core/calendar_event_created/user_created_calendar_event/data.json:1]: formatting

File: tests/core/calendar_event_created/user_created_calendar_event/statements.json
Comment (milt) [inline tests/core/calendar_event_created/user_created_calendar_event/statements.json:1]: formatting

File: tests/core/calendar_event_updated/user_updated_calendar_event/statements.json
Comment (milt) [inline tests/core/calendar_event_updated/user_updated_calendar_event/statements.json:1]: formatting

File: src/transformer/events/core/badge_revoked.php
Comment (invaliduser) [reply src/transformer/events/core/badge_revoked.php:1]: this was removal of debug code

File: src/transformer/events/core/calendar_event_created.php (hunk 2 above)
Comment (invaliduser) [reply src/transformer/events/core/calendar_event_created.php]: removed

File: src/transformer/get_event_function_map.php (hunk 3 above)
Comment (invaliduser) [reply src/transformer/get_event_function_map.php]: removed

File: src/transformer/get_event_function_map.php (hunk 4 above)
Comment (invaliduser) [reply src/transformer/get_event_function_map.php:103]: They were added before that, but this commit moves them up
https://github.com/yetanalytics/moodle-logstore_xapi/blame/master/src/transformer/get_event_function_map.php

File: tests/core/calendar_event_created/user_created_calendar_event/statements.json
Comment (invaliduser) [reply tests/core/calendar_event_created/user_created_calendar_event/statements.json:1]: fixed

File: tests/core/calendar_event_updated/user_updated_calendar_event/statements.json
Comment (invaliduser) [reply tests/core/calendar_event_updated/user_updated_calendar_event/statements.json:1]: fixed

File: tests/core/calendar_event_created/user_created_calendar_event/statements.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+		}
+	    },
+	    "extensions":{
+		"http://lrs.learninglocker.net/define/extensions/info": {
+		    "http://moodle.org": "1.0.0",
+		    "https://github.com/xAPI-vle/moodle-logstore_xapi": "0.0.0-development",
+		    "event_name": "\\core\\event\\calendar_event_created",
+		    "event_function": "\\src\\transformer\\events\\core\\calendar_event_created"
+		}
+	    }
```
Comment (milt) [inline tests/core/calendar_event_created/user_created_calendar_event/statements.json]: stuff like this is still improperly formatted

File: tests/core/calendar_event_created/user_created_calendar_event/statements.json
Comment (milt) [reply tests/core/calendar_event_created/user_created_calendar_event/statements.json:1]: Go ahead and check it again, there is still nesting that doesn't indent

Comment (invaliduser) [reply tests/core/calendar_event_created/user_created_calendar_event/statements.json:1]: are we talking about the same file?

image

Comment (milt) [reply tests/core/calendar_event_created/user_created_calendar_event/statements.json:1]: Yes (let's discuss further on slack)

File: tests/core/calendar_event_created/user_created_calendar_event/statements.json [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                  },
+                  "name" : {
+                     "en" : "test_course_name"
+                  },
+                  "type" : "https://w3id.org/xapi/cmi5/activitytype/course"
+               },
+               "id" : "http://www.example.org/course/view.php?id=1",
+               "objectType" : "Activity"
+            }
+         },
```
Comment (milt) [inline tests/core/calendar_event_created/user_created_calendar_event/statements.json]: So I might have missed this on the last PR and now need to go back and check, but this is invalid xAPI. The properties of the contextActivities object are arrays

File: tests/core/calendar_event_created/user_created_calendar_event/statements.json [hunk 7]
```diff
@@ -0,0 +1,65 @@
+[
+   {
+      "actor" : {
+         "account" : {
+            "homePage" : "http://www.example.org",
+            "name" : 1
```
Comment (milt) [inline tests/core/calendar_event_created/user_created_calendar_event/statements.json]: account name needs to be a string

Comment (invaliduser) [reply tests/core/calendar_event_created/user_created_calendar_event/statements.json]: changed

**Review by milt — APPROVED** [review]

File: src/transformer/events/core/calendar_event_created.php [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        ],
+        'context' => [
+            'extensions' => utils\extensions\base($config, $event, $course),
+            'contextActivities' => [
+                'category' => [[
+                    'id' => $config['app_url'],
+                    'objectType' => 'Activity',
+                    'definition' => [
+                        'name' => [
+                            'en' => 'EDLM Moodle LMS'
```
Comment (milt) [inline src/transformer/events/core/calendar_event_created.php]: this should probably take a name from config if it is there

File: tests/core/calendar_event_created/user_created_calendar_event/user_created_calendar_event_test.php [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+namespace logstore_xapi\core\calendar_event_created\user_created_calendar_event;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+
+require_once($CFG->dirroot . '/admin/tool/log/store/xapi/tests/xapi_test_case.php');
+
+/**
+ * Unit test for mod_quiz attempt submitted (multichoice set with choices) event.
```
Comment (milt) [inline tests/core/calendar_event_created/user_created_calendar_event/user_created_calendar_event_test.php]: incorrect

File: tests/core/calendar_event_updated/user_updated_calendar_event/user_updated_calendar_event_test.php [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+namespace logstore_xapi\core\calendar_event_created\user_created_calendar_event;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+
+require_once($CFG->dirroot . '/admin/tool/log/store/xapi/tests/xapi_test_case.php');
+
+/**
+ * Unit test for mod_quiz attempt submitted (multichoice set with choices) event.
```
Comment (milt) [inline tests/core/calendar_event_updated/user_updated_calendar_event/user_updated_calendar_event_test.php]: incorrect

**Review by cliffcaseyyet — APPROVED** [review]
Milt has some outstanding comments in here but otherwise i approve.

File: src/transformer/events/core/calendar_event_created.php (hunk 8 above)
Comment (milt) [reply src/transformer/events/core/calendar_event_created.php]: as discussed on slack, use site. Look at my example here https://github.com/yetanalytics/moodle-logstore_xapi/pull/11/files

File: src/transformer/events/core/calendar_event_created.php [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Transform for course viewed event.
```
Comment (milt) [inline src/transformer/events/core/calendar_event_created.php]: copy pasta

File: src/transformer/events/core/calendar_event_updated.php [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+ * @license   https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace src\transformer\events\core;
+
+use src\transformer\utils as utils;
+use src\transformer\utils\get_activity as activity;
+
+/**
+ * Transformer for Calendar Event Created.
```
Comment (milt) [inline src/transformer/events/core/calendar_event_updated.php]: should be updated

File: tests/core/calendar_event_deleted/user_deleted_calendar_event/user_deleted_calendar_event_test.php [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+namespace logstore_xapi\core\calendar_event_deleted\user_deleted_calendar_event;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+
+require_once($CFG->dirroot . '/admin/tool/log/store/xapi/tests/xapi_test_case.php');
+
+/**
+ * Unit test for mod_quiz attempt submitted (multichoice set with choices) event.
```
Comment (milt) [inline tests/core/calendar_event_deleted/user_deleted_calendar_event/user_deleted_calendar_event_test.php]: pasta

File: tests/core/calendar_event_created/user_created_calendar_event/user_created_calendar_event_test.php (hunk 9 above)
Comment (invaliduser) [reply tests/core/calendar_event_created/user_created_calendar_event/user_created_calendar_event_test.php]: changed

File: tests/core/calendar_event_updated/user_updated_calendar_event/user_updated_calendar_event_test.php (hunk 10 above)
Comment (invaliduser) [reply tests/core/calendar_event_updated/user_updated_calendar_event/user_updated_calendar_event_test.php]: changed

File: src/transformer/events/core/calendar_event_updated.php (hunk 12 above)
Comment (invaliduser) [reply src/transformer/events/core/calendar_event_updated.php]: changed

File: tests/core/calendar_event_deleted/user_deleted_calendar_event/user_deleted_calendar_event_test.php (hunk 13 above)
Comment (invaliduser) [reply tests/core/calendar_event_deleted/user_deleted_calendar_event/user_deleted_calendar_event_test.php]: changed

**Pull request merged**